components/style/traversal.rs

```diff
@@ -535,7 +546,9 @@ fn compute_style<E, D>(_traversal: &D,
     }
 
     // FIXME(bholley): Compute this accurately from the call to CalcStyleDifference.
-    let inherited_styles_changed = true;
+    let inherited_styles_changed = data.get_restyle_mut().map_or(true, |r| {
+        !r.damage.is_empty()
```

Comment:
Please add a comment that this still is much more conservative than we want (since we want to be checking aEqualStructs), but should still be correct.